CMakeLists.txt

```diff
@@ -16,8 +16,10 @@ endif ()
 ## Main library build
 ##
 
-set(CMAKE_CXX_VISIBILITY_PRESET hidden)
-set(CMAKE_VISIBILITY_INLINES_HIDDEN YES)
+if (BUILD_SHARED_LIBS)
+    set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+    set(CMAKE_VISIBILITY_INLINES_HIDDEN YES)
+endif()

```

Comment:
I'm still not clear on why hiding symbols for static libraries is a problem. It seems to me that the only necessary change is the one in `tinyxml2.h`.

Reply:
Discussed in https://github.com/leethomason/tinyxml2/pull/922#discussion_r1008787957.

Do you have an example that builds a shared library file that uses tinyxml2 internally but doesn't export the symbols of tinyxml2 without this change?